crates/wasmtime/src/runtime/store.rs

```diff
@@ -1523,7 +1535,8 @@ impl StoreOpaque {
         &self.runtime_limits as *const VMRuntimeLimits as *mut VMRuntimeLimits
     }
 
-    pub unsafe fn insert_vmexternref_without_gc(&mut self, r: VMExternRef) {
+    #[cfg(feature = "gc")]
```

Comment:
In the interest of reducing `#[cfg]` as much as possible (or at least centralizing it) could this and other helper methods be preserved here? That way in `wasmtime-runtime` itself it would have the `match x {}` to assert unreachability (or some method on `VMExternRef` itself)

Reply:
The problem with this specific case is that we store the `VMExternRefActivationsTable` directly in the store, so if it is uninhabited, then the whole `Store` becomes uninhabited.

Reply:
We could make it an `Option`, but then we enable the potential for accidentally setting it to `None` when GC *is* enabled. I think I'd rather leave these handful of `cfg`s in comparison to that.

Reply:
I guess we could have a `GcOnly<T>` type like

```rust
enum GcOnly<T> {
    #[cfg(feature = "gc")]
    inner: T,
}

impl<T> GcOnly<T> {
    pub fn with<U>(&self, f: impl FnMut(&T) -> U) -> Option<U> {
        #[cfg(feature = "gc")]
        return Some(f(&self.inner));
        #[cfg(not(feature = "gc"))]
        return None;
    }
}
```

But that seems like overkill maybe?

Reply:
I'd personally prefer to avoid `GcOnly<T>`, I feel like those sorts of wrappers haven't worked out well in the past or require otherwise just as much `#[cfg]` traffic. For `VMExternRefActivationsTable` I'd expect that'd become an empty struct rather than uninhabited because there are no references to track?

Reply:
Ah yeah I could make that change, good idea.

Reply:
Done!